homeassistant/components/climacell/config_flow.py

```diff
@@ -0,0 +1,195 @@
+"""Config flow for ClimaCell integration."""
+import logging
+from typing import Any, Dict
+
+from pyclimacell import ClimaCell
+from pyclimacell.const import REALTIME
+from pyclimacell.exceptions import (
+    CantConnectException,
+    InvalidAPIKeyException,
+    RateLimitedException,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.components.air_quality import DOMAIN as AIR_QUALITY_DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_registry import (
+    async_entries_for_config_entry,
+    async_get_registry,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    CHINA,
+    CONF_AQI_COUNTRY,
+    CONF_FORECAST_TYPE,
+    CONF_TIMESTEP,
+    DAILY,
+    DEFAULT_AQI_COUNTRY,
+    DEFAULT_FORECAST_TYPE,
+    DEFAULT_NAME,
+    DEFAULT_TIMESTEP,
+    DISABLE_FORECASTS,
+    DOMAIN,
+    HOURLY,
+    NOWCAST,
+    USA,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_config_schema(
+    hass: core.HomeAssistant, input_dict: Dict[str, Any] = None
+) -> vol.Schema:
+    """
+    Return schema defaults for init step based on user input/config dict.
+
+    Retain info already provided for future form views by setting them as
+    defaults in schema.
+    """
+    if input_dict is None:
+        input_dict = {}
+
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=input_dict.get(CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(CONF_API_KEY, default=input_dict.get(CONF_API_KEY)): str,
+            vol.Inclusive(
+                CONF_LATITUDE,
+                "location",
+                default=input_dict.get(CONF_LATITUDE, hass.config.latitude),
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_LONGITUDE,
+                "location",
+                default=input_dict.get(CONF_LONGITUDE, hass.config.longitude),
+            ): cv.longitude,
+            vol.Optional(
+                CONF_FORECAST_TYPE,
```

Comment:
Can we create all forecast entity types and disable all but one instead of having this option?
https://developers.home-assistant.io/docs/entity_registry_disabled_by